src/main/scala/zio/prelude/Identity.scala

```diff
@@ -817,4 +817,24 @@ trait IdentitySyntax {
     def identity(implicit id: Identity[A]): A =
       id.identity
   }
+
+  /**
+   * Provides syntax for combining values in a `ParIterable` which is thus done in a parallel manner.
+   */
+  implicit class IdentityParIterableOps[A](private val p: collection.parallel.immutable.ParIterable[A]) {
+
+    /**
+     * Associatively combines the values in a parallel manner, returning the `identity` element if empty,
+     * while blocking the thread.
+     */
+    def reduceIdentityBlocking(implicit identity: Identity[A]): A =
+      p.fold(identity.identity)(identity.combine(_, _))
+
+    /**
+     * Returns an effect, that associatively combines the values in a parallel manner, returning the `identity` element if empty,
+     * while ensuring the current thread isn't blocked.
+     */
+    def reduceIdentity(implicit identity: Identity[A]): zio.RIO[zio.blocking.Blocking, A] =
```

Comment:
AFAIK, ZIO Prelude is moving to having ZIO proper only as an _optional_ dependency. In light of that, do you think we should change this signature?
Signature like this
```scala
    def reduceIdentity(blocker: zio.blocking.Blocking)(implicit identity: Identity[A]): zio.Task[A]
```
would "force" the caller to supply a ZIO object. That could prevent runtime error in cases ZIO is not on classpath (which could happen, since it would be an optional dependency).

@jdegoes @adamgfraser

Reply:
I think that makes it more painful to use without any real gain. A user can always add ZIO if they want to use these optional methods.